code/modules/transport/tram/tram_structures.dm

```diff
@@ -0,0 +1,579 @@
+/**
+ * the tram has a few objects mapped onto it at roundstart, by default many of those objects have unwanted properties
+ * for example grilles and windows have the atmos_sensitive element applied to them, which makes them register to
+ * themselves moving to re register signals onto the turf via connect_loc. this is bad and dumb since it makes the tram
+ * more expensive to move.
+ *
+ * if you map something on to the tram, make SURE if possible that it doesnt have anything reacting to its own movement
+ * it will make the tram more expensive to move and we dont want that because we dont want to return to the days where
+ * the tram took a third of the tick per movement when its just carrying its default mapped in objects
+ */
+
+/obj/structure/grille/tram/Initialize(mapload)
+	. = ..()
+	RemoveElement(/datum/element/atmos_sensitive, mapload)
+	//atmos_sensitive applies connect_loc which 1. reacts to movement in order to 2. unregister and register signals to
+	//the old and new locs. we dont want that, pretend these grilles and windows are plastic or something idk
+
+/obj/structure/tram/Initialize(mapload, direct)
+	. = ..()
+	RemoveElement(/datum/element/atmos_sensitive, mapload)
+
+/obj/structure/tram
+	name = "tram wall"
+	desc = "A lightweight titanium composite structure with titanium silicate panels."
+	icon = 'icons/obj/tram/tram_structure.dmi'
+	icon_state = "tram-part-0"
+	base_icon_state = "tram-part"
+	max_integrity = 150
+	layer = TRAM_WALL_LAYER
+	density = TRUE
+	opacity = FALSE
+	anchored = TRUE
+	flags_1 = PREVENT_CLICK_UNDER_1
+	armor_type = /datum/armor/tram_structure
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = SMOOTH_GROUP_TRAM_STRUCTURE
+	canSmoothWith = SMOOTH_GROUP_TRAM_STRUCTURE
+	can_be_unanchored = FALSE
+	can_atmos_pass = ATMOS_PASS_DENSITY
+	explosion_block = 3
+	receive_ricochet_chance_mod = 1.2
+	rad_insulation = RAD_MEDIUM_INSULATION
+	var/state = TRAM_SCREWED_TO_FRAME
+	var/mineral = /obj/item/stack/sheet/titaniumglass
+	var/mineral_amount = 2
+	var/tram_wall_type = /obj/structure/tram
+	var/girder_type = /obj/structure/girder/tram
+	var/mutable_appearance/damage_overlay
+	var/break_sound = SFX_SHATTER
+	var/knock_sound = 'sound/effects/glassknock.ogg'
+	var/bash_sound = 'sound/effects/glassbash.ogg'
+
+/obj/structure/tram/solid
+	desc = "A lightweight titanium composite structure with tinted titanium silicate panels."
+	opacity = TRUE
```

Comment:
dont put this or anything that has opacity on the roundstart tram, it causes SSlighting to spam lighting updates every time the tram moves a tile.
![Screenshot_3122](https://github.com/tgstation/tgstation/assets/15794172/9eac4798-6a3f-4acf-9daa-941b69e1136b)

Reply:
ah, didn't know opacity did that. I'll go over it.